### PR TITLE
Stack refactor

### PR DESCRIPTION
**Pull request by etsauer**
#### What does this PR do?
Heavy refactoring of our heat templates to have them be generated by ansible templates.

This will relieve us of some of the technical debt we've built up by borrowing templates from another project and opens the door for HA masters and other more advanced provisioning we would like to be able to do.

#### How should this be manually tested?
```
ansible-playbook -i /root/code/casl-ansible-inventories/l1.casl.rht-labs.com.d/inventory/ code/casl-ansible/playbooks/openshift/end-to-end.yml
```

#### Is there a relevant Issue open for this?
https://github.com/redhat-cop/casl-ansible/issues/36 and https://github.com/redhat-cop/casl-ansible/issues/12

#### Who would you like to review this?
cc: @redhat-cop/casl

### CODE REVIEW

File: inventory/sample.casl.example.com.d/inventory/group_vars/all.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+ansible_user: openshift
+ansible_become: True
+
+# OpenStack Provisioning Variables
+openstack_ssh_public_key: "esauer-util3"
```
Comment (oybed) [inline inventory/sample.casl.example.com.d/inventory/group_vars/all.yml]: Consider replacing this fixed value with:
```
# use the "-e" option to set "openstack_ssh_public_key"
```

File: inventory/sample.casl.example.com.d/inventory/group_vars/all.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+ansible_user: openshift
+ansible_become: True
+
+# OpenStack Provisioning Variables
+openstack_ssh_public_key: "esauer-util3"  
+openstack_default_image_name: "rhel-guest-image-7.3-35.x86_64"
+openstack_default_flavor: "m1.large"
```
Comment (oybed) [inline inventory/sample.casl.example.com.d/inventory/group_vars/all.yml]: Please change to `m1.medium`

File: inventory/sample.casl.example.com.d/inventory/group_vars/all.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+openstack_subnet_prefix: 192.168.99
+openstack_nameservers: 10.9.48.31
+openstack_num_nodes: 2
+openstack_num_infra: 1
+dns_domain: "casl.example.com"
+cluster_id: "casl.example.com"
+public_dns_forwarder: 10.9.48.31
+env_id: "sample"
+#
+docker_volume_size: "50"
```
Comment (oybed) [inline inventory/sample.casl.example.com.d/inventory/group_vars/all.yml]: Please change to `10` or even `5`

File: inventory/sample.casl.example.com.d/inventory/group_vars/all.yml (hunk 1 above)
Comment (sabre1041) [inline inventory/sample.casl.example.com.d/inventory/group_vars/all.yml]: Please replace `esauer-util3` with a generic name

Comment (sabre1041) [inline inventory/sample.casl.example.com.d/inventory/group_vars/all.yml]: Replace `esauer-util3` with a generic name

File: roles/openstack-stack/templates/heat_stack_server.yaml.j2 [hunk 4]
```diff
@@ -83,6 +89,11 @@ parameters:
       - range: { min: 1, max: 1024 }
         description: must be between 1 and 1024 Gb.
 
+  node_labels:
+    type: json
+    description: OpenShift Node Labels
+    default: {"region": "default" }
```
Comment (oybed) [inline roles/openstack-stack/templates/heat_stack_server.yaml.j2]: What happened to `region: infra`?

**Comment by oybed** [comment]
@etsauer we had a request to include the default node selector (i.e.: `osm_default_node_selector`) to avoid having regular apps land on the infra nodes. Do you think that can be included as part of this PR, or should we do a follow-up PR?

https://github.com/openshift/openshift-ansible/blob/master/inventory/byo/hosts.ose.example#L260

File: roles/openstack-stack/templates/heat_stack_server.yaml.j2 (hunk 4 above)
Comment (etsauer) [inline roles/openstack-stack/templates/heat_stack_server.yaml.j2]: We apply `region:infra` to the infra nodes, but heat requires a default value for the parameter. I didn't want to make it something that actually has meaning.

**Comment by etsauer** [comment]
@oybed @sabre1041 updated per your comments.

**Comment by oybed** [comment]
LGTM

**Pull request merged**